### PR TITLE
Do canonical headers attempt #2

### PR DESCRIPTION
**Pull request by motobob**
in #3, #4, #5 is was found out that `partial_upload_chunked` test started to fail.

this PR is open to figure out if the change is still required and if so fix the remainder of things

### CODE REVIEW

**Comment by motobob** [comment]
it's now canonical again. tests fixed.
@nalundgaard @fp FYI.

i'm going to merge this, tag it, hex it and re-pin erlcloud to fix it and merge too.

File: src/lhttpc_lib.erl [hunk 1]
```diff
@@ -73,30 +75,44 @@ header_value(Hdr, Hdrs) ->
 %% @doc
 %% Returns the value associated with the `Header' in `Headers'.
 %% `Header' must be a lowercase string, since every header is mangled to
-%% check the match.  If no match is found, `Default' is returned.
+%% check the match.  `Headres' must be canonical.
```
Comment (nalundgaard) [inline src/lhttpc_lib.erl]: Headres?

**Comment by nalundgaard** [comment]
Oh this is a pure revert. IMO that is the right thing to do, rather than eliminating it just to make the tests that were added in a separate fork work right. I do suspect that indicates a bug of some sort, but header normalization seems pretty important.

**Comment by motobob** [comment]
this is a revert since without header normalisation lots of things break.

**Comment by nalundgaard** [comment]
👍

**Comment by motobob** [comment]
verified manually on a top level app.
```erlang
> rp(catch erlcloud_s3:s3_request(Cfg, get, "x-account-test-ireland", "/", "", [], <<>>, [])). 
{'EXIT',{{aws_error,{http_error,307,"Temporary Redirect",
                                <<"<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>TemporaryRedirect</Code><Message>Please re-send this request to the specified temporary endpoint. Continue to use the original request endpoint for future requests.</Message><Bucket>x-account-test-ireland</Bucket><Endpoint>x-account-test-ireland.s3-eu-west-1.amazonaws.com</Endpoint><RequestId>ACE88403747A6F90</RequestId><HostId>sxROxXHxoj3StibUYFgRm5m5DmQspqU1e0mggXtpvhhTmjgUKkpYo4OLlCjBpFj/kMtMc8+4Uc4=</HostId></Error>">>,
                                [{"server","AmazonS3"},
                                 {"date","Fri, 30 Dec 2016 21:12:02 GMT"},
                                 {"transfer-encoding","chunked"},
                                 {"content-type","application/xml"},
                                 {"location",
                                  "https://x-account-test-ireland.s3-eu-west-1.amazonaws.com/"},
                                 {"x-amz-id-2",
                                  "sxROxXHxoj3StibUYFgRm5m5DmQspqU1e0mggXtpvhhTmjgUKkpYo4OLlCjBpFj/kMtMc8+4Uc4="},
                                 {"x-amz-request-id","ACE88403747A6F90"},
                                 {"x-amz-bucket-region","eu-west-1"}]}},
         [{erlcloud_s3,s3_request,8,
                       [{file,"src/erlcloud_s3.erl"},{line,1340}]},
```

**Pull request merged**